openedx/features/enterprise_support/tests/test_api.py

```diff
@@ -159,6 +168,54 @@ def test_enterprise_api_client_with_user(self, mock_jwt_builder):
         """
         self._assert_api_client_with_user(EnterpriseApiClient, mock_jwt_builder)
 
+    @httpretty.activate
+    @mock.patch('openedx.features.enterprise_support.api.create_jwt_for_user')
+    def test_enterprise_api_client_with_user_post_enrollment(self, mock_jwt_builder):
+        """
+        Verify that enterprise API client uses the provided user to
+        authenticate and access enterprise API.
+        """
+        api_client = self._assert_api_client_with_user(EnterpriseApiClient, mock_jwt_builder)
+        setattr(api_client.client, 'enterprise-course-enrollment', mock.Mock())
+
+        username = 'spongebob'
+        course_id = 'burger-flipping-101'
+        consent_granted = True
+
+        api_client.post_enterprise_course_enrollment(username, course_id, consent_granted)
+
+        mock_endpoint = getattr(api_client.client, 'enterprise-course-enrollment')
+        mock_endpoint.post.assert_called_once_with(data={
+            'username': username,
+            'course_id': course_id,
+            'consent_granted': consent_granted,
+        })
+
+    @httpretty.activate
+    @mock.patch('openedx.features.enterprise_support.api.create_jwt_for_user')
+    def test_enterprise_api_client_with_user_post_enrollment_http_error(self, mock_jwt_builder):
+        """
+        Verify that enterprise API client uses the provided user to
+        authenticate and access enterprise API.
+        """
+        api_client = self._assert_api_client_with_user(EnterpriseApiClient, mock_jwt_builder)
+        setattr(api_client.client, 'enterprise-course-enrollment', mock.Mock())
+        mock_endpoint = getattr(api_client.client, 'enterprise-course-enrollment')
+        mock_endpoint.post.side_effect = HttpClientError
+
+        username = 'spongebob'
+        course_id = 'burger-flipping-101'
+        consent_granted = True
```

Comment:
I don't feel that strongly about it since I feel like being super DRY in tests often couples things together very tightly, but this and `test_enterprise_api_client_with_user_post_enrollment` are very similar and could probably share some things. I leave the decision to your discretion though